src/Runtime/PyExecutionSessionBase.cpp

```diff
@@ -21,6 +22,36 @@ SUPPRESS_WARNINGS_POP
 
 #include "PyExecutionSessionBase.hpp"
 
+namespace pybind11 {
+namespace detail {
+
+// Note: Since float16 is not a builtin type in C++, we register
+// onnx_mlir::float_16 as numpy.float16.
+// Ref: https://github.com/pybind/pybind11/issues/1776
+//
+// This implementation is copied from https://github.com/PaddlePaddle/Paddle
```

Comment:
I assume that since you included this code in a file with Apache license header, the code copied here is compatible with all its requirements.